Add codes.merritt.Nyrna

Merrit opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/Merrit/nyrna
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

hfiguiere commented on codes.merritt.Nyrna.yml:
```diff
@@ -0,0 +1,47 @@
+---
+app-id: codes.merritt.Nyrna
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
```suggestion
runtime-version: '22.08'
```

hfiguiere commented on codes.merritt.Nyrna.yml:
```diff
@@ -0,0 +1,47 @@
+    path: codes.merritt.Nyrna.metainfo.xml
+- name: keybinder3
+  buildsystem: autotools
+  only-arches:
+  - x86_64
```
```suggestion
```

hfiguiere commented on codes.merritt.Nyrna.yml:
```diff
@@ -0,0 +1,47 @@
+  buildsystem: autotools
+  only-arches:
+  - x86_64
+  post-install: install -Dm644 COPYING /app/share/licenses/keybinder-3/COPYING
```
```suggestion
  post-install: 
    - install -Dm644 COPYING /app/share/licenses/keybinder-3/COPYING
```

hfiguiere replied:
```
(flatpak-builder:3): Json-WARNING **: 07:22:14.749: Failed to deserialize "post-install" property of type "GStrv" for an object of type "BuilderModule"
```

hfiguiere commented on codes.merritt.Nyrna.yml:
```diff
@@ -0,0 +1,47 @@
+- --talk-name=org.freedesktop.Flatpak # Access host processes
+modules:
+# Nyrna
+- name: nyrna
```
since this is the main module it should be last.

hfiguiere commented on codes.merritt.Nyrna.yml:
```diff
@@ -0,0 +1,47 @@
+    # during build resolve by running `git config --global --add protocol.file.allow always`
+    disable-shallow-clone: true
+  - type: file
+    path: codes.merritt.Nyrna.metainfo.xml
```
this file should be managed upstream for a future release.

hfiguiere replied:
actually it seems to be usptream already.

hfiguiere commented:
bot, build codes.merritt.Nyrna

hfiguiere commented on codes.merritt.Nyrna.yml:
```diff
@@ -0,0 +1,47 @@
+    url: https://github.com/kupferlauncher/keybinder/releases/download/keybinder-3.0-v0.3.2/keybinder-3.0-0.3.2.tar.gz
+    sha256: e6e3de4e1f3b201814a956ab8f16dfc8a262db1937ff1eee4d855365398c6020
+# libappindicator
+- shared-modules/libappindicator/libappindicator-gtk3-12.10.json
```
`shared-modules` is missing.

Merrit commented:
Made changes :+1:

Merrit commented:
bot, build codes.merritt.Nyrna

Merrit commented:
Looks like the build service had a network error?

Merrit commented:
bot, build codes.merritt.Nyrna

dginovker commented:
![image](https://user-images.githubusercontent.com/32943174/212162438-ee4be2d0-9582-4e9f-b38a-44b2e1af8218.png)

It'll be up soon

barthalion commented:
bot, build codes.merritt.Nyrna

barthalion commented:
You will need to submit your app to https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/staticfiles/exceptions.json. I assume you need access to flatpak-spawn to run SIGSTOP on arbitrary host processes?

Merrit commented:
Exactly right, yes. I created a PR to add the exception, thank you for directing me to it! :)

Merrit commented:
bot, build codes.merritt.Nyrna

Merrit commented:
Seems to install and run as expected :+1:

barthalion commented:
I get the "No windows that can be suspended" message on Fedora 37 Silverblue, despite having multiple windows open.

Merrit commented:
Is that on Wayland? Only supports X11 at the moment unfortunately.

hfiguiere commented on codes.merritt.Nyrna.yml:
```diff
@@ -0,0 +1,46 @@
+separate-locales: false
+finish-args:
+- --share=ipc
+- --socket=fallback-x11
+- --socket=wayland
```
you just said it only support x11. so that's incorrect then.

Merrit replied:
Thank you for pointing that out! <3

Merrit commented:
bot, build codes.merritt.Nyrna

Merrit commented:
Dependency check works inside Flatpak now. Also added a warning if a user tries to run it under Wayland, and changed the config to only specify X11 support. :+1:

barthalion commented:
/merge
